tests/req_tests.rs

```diff
@@ -89,3 +89,103 @@ fn test_add_route() {
 fn _testing_callback(req: Request) -> Response {
     servo::http::ok(String::from("test me"), CONTENT_TYPE::TEXT_HTML)
 }
+
+// Tests for parsing the query string to params and args
+#[test]
+fn test_params_from_route() {
+    let mut request = Request::new().with_route("/test/route/param?arg1=1&arg2=2".to_string());
+    let query: Vec<&str> = ["param", "arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    assert_eq!(request.get_params(), ["param".to_string()].to_vec())
+}
+
+#[test]
+fn test_multiple_params_from_route() {
+    let mut request = Request::new()
+                              .with_route("/test/route/param1?param2?arg1=1&arg2=2".to_string());
+    let query: Vec<&str> = ["param1", "param2", "arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    assert_eq!(request.get_params() , ["param1".to_string(), "param2".to_string()].to_vec())
+}
+
+#[test]
+fn test_empty_params_from_route() {
+    let mut request = Request::new()
+                              .with_route("/test/route/arg1=1&arg2=2".to_string());
+    let query: Vec<&str> = ["arg1=1&arg2=2"].to_vec();
+    request = request.params_from_route(query);
+    let compare: Vec<String> = Vec::new();
+    assert_eq!(request.get_params() , compare)
+}
+
+#[test]
+fn test_args_from_route() {
```

Comment:
This looks as if it's parsing out the same thing as the param parsing is. Is the arg parsing just duplicating the param parsing?